global_body_planner/include/global_body_planner/global_body_planner_test_fixture.h

```diff
@@ -0,0 +1,119 @@
+#ifndef GLOBAL_BODY_PLANNER_TEST_H
+#define GLOBAL_BODY_PLANNER_TEST_H
+
+#include <gtest/gtest.h>
+#include <quad_utils/fast_terrain_map.h>
+#include <quad_utils/ros_utils.h>
+#include <ros/ros.h>
+
+#include <grid_map_core/grid_map_core.hpp>
+#include <grid_map_ros/GridMapRosConverter.hpp>
+#include <grid_map_ros/grid_map_ros.hpp>
+
+#include "global_body_planner/planner_class.h"
+#include "global_body_planner/planning_utils.h"
+#include "global_body_planner/rrt_connect.h"
+
+//! A test fixture for the global body planning class
+/*!
+   GlobalBodyPlannerTest is a container for all of the logic utilized in the
+   global body planning node. This algorithm requires an height map of the
+   terrain as a GridMap message type, and will publish the global body plan as a
+   BodyPlan message over a topic. It will also publish the discrete states used
+   by the planner (from which the full path is interpolated).
+*/
+class GlobalBodyPlannerTestFixture : public ::testing::Test {
+ public:
+  /**
+   * @brief Constructor for GlobalBodyPlannerTestFixture Class
+   * @return Constructed object of type GlobalBodyPlannerTestFixture
+   */
+  GlobalBodyPlannerTestFixture() : planner_(FORWARD) {
+    grid_map::GridMap map({"z_inpainted", "z_smooth", "normal_vectors_x",
+                           "normal_vectors_y", "normal_vectors_z",
+                           "smooth_normal_vectors_x", "smooth_normal_vectors_y",
+                           "smooth_normal_vectors_z", "traversability"});
+    map.setGeometry(grid_map::Length(10, 10), 0.05);
+    terrain_grid_map_ = map;
+
+    double slope = 0;
+    updateTerrainSlope(slope);
+
+    // Create planner and configuration
+    ros::NodeHandle nh;
+    planner_config_.loadParamsFromServer(nh);
+  }
+
+  /**
+   * @brief Constructor for GlobalBodyPlannerTestFixture Class
+   * @param[in] grade Grade of the new terrain in x direction (rise/run)
+   */
+  void updateTerrainSlope(double grade) {
```

Comment:
If precision is not critical float is better to use instead of double, but if this is how we have it elsewhere then I guess its ok. Not a big deal